symbolic-debuginfo/src/base.rs

```diff
@@ -446,31 +446,62 @@ impl<'d> FromIterator<Symbol<'d>> for SymbolMap<'d> {
 #[derive(Clone, Default, Eq, PartialEq)]
 pub struct FileInfo<'data> {
     /// The file's basename.
-    pub name: &'data [u8],
+    name: Cow<'data, [u8]>,
     /// Path to the file.
-    pub dir: &'data [u8],
+    dir: Cow<'data, [u8]>,
 }
 
 impl<'data> FileInfo<'data> {
+    /// Creates a `FileInfo` with a given directory and the file name.
+    #[cfg(feature = "dwarf")]
+    pub fn new(dir: Cow<'data, [u8]>, name: Cow<'data, [u8]>) -> Self {
+        FileInfo { name, dir }
+    }
+
     /// Creates a `FileInfo` from a joined path by trying to split it.
     #[cfg(any(feature = "breakpad", feature = "ms", feature = "sourcebundle"))]
-    pub(crate) fn from_path(path: &'data [u8]) -> Self {
+    pub fn from_path(path: &'data [u8]) -> Self {
+        let (dir, name) = symbolic_common::split_path_bytes(path);
+
+        FileInfo {
+            name: Cow::Borrowed(name),
+            dir: match dir {
+                Some(dir) => Cow::Borrowed(dir),
+                None => Cow::default(),
+            },
+        }
+    }
+
+    /// Creates a `FileInfo` from a joined path by trying to split it.
+    #[cfg(feature = "ppdb")]
+    pub(crate) fn from_path_clone(path: &[u8]) -> Self {
```

Comment:
```suggestion
    pub(crate) fn from_path_owned(path: &[u8]) -> Self {
```

The documentation could also briefly point out the difference between this and `from_path`.

Reply:
done